File: 9. Sum of the numbers in a String/code.py
def sum_of_numbers(s):
    num = 0
    current_num = ""
    for char in s:
        if char.isdigit():
            current_num += char
        elif current_num:
            num += int(current_num)
            current_num = ""
    if current_num:
        num += int(current_num)
    return num

input_str = input("Enter a string: ")
print("Sum of numbers in the string:", sum_of_numbers(input_str))

File: 9. Sum of the numbers in a String/code.cpp
#include <iostream>
#include <string>
using namespace std;

int sumOfNumbers(string s) {
    int sum = 0;
    string currentNum = "";

    for (char c : s) {
        if (isdigit(c)) {
            currentNum += c;
        } else if (!currentNum.empty()) {
            sum += stoi(currentNum);
            currentNum = "";
        }
    }

    if (!currentNum.empty()) {
        sum += stoi(currentNum);
    }

    return sum;
}

int main() {
    string input_str;
    cout << "Enter a string: ";
    getline(cin, input_str);
    cout << "Sum of numbers in the string: " << sumOfNumbers(input_str) << endl;
    return 0;
}
